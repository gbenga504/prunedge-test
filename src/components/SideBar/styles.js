export const styles = theme => ({
  container: {
    position: "absolute",
    left: 0,
    top: 0,
    height: "100%",
    width: 265,
    paddingBottom: 76,
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    background: "#FFFFFF",
    boxShadow: "4px 0px 24px #E3E3E3"
  },
  topSectionContainer: {
    display: "flex",
    flexDirection: "column",
    marginTop: 95
  },
  userDetailsContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  },
  avatarContainer: {
    borderRadius: 20,
    border: "2px solid #7396FD",
    background: "#fff",
    height: 85,
    width: 85,
    boxSizing: "unset",
    "& img": {
      borderRadius: 20,
      height: "calc(100% - 8px)",
      width: "calc(100% - 8px)",
      boxSizing: "unset",
      margin: 4
    }
  },
  userName: {
    fontSize: 14,
    fontFamily: "Lato",
    color: "#5F5C7F",
    fontWeight: 700,
    marginTop: 16
  },
  role: {
    fontSize: 12,
    color: "#999",
    fontWeight: 400,
    fontFamily: "Lato",
    marginTop: 5
  },
  department: {
    fontSize: 12,
    color: "#999",
    fontFamily: "Lato",
    fontWeight: 400,
    opacity: 0.7,
    marginTop: 5
  },
  navItem: {
    width: "100%",
    height: 48,
    display: "flex",
    alignItems: "center",
    paddingLeft: 40,
    cursor: "pointer"
  },
  navItemActive: {
    "&::before": {
      content: "''",
      position: "absolute",
      width: 5,
      height: 48,
      left: 0,
      borderRadius: "0px 3px 3px 0px",
      background: "#0C66FF"
    }
  },
  icon: {
    fontSize: 26,
    color: "#999",
    marginRight: 10
  },
  iconActive: {
    color: "#182C4F"
  },
  navItemText: {
    fontSize: 14,
    fontFamily: "Lato",
    color: "#999",
    textTransform: "capitalize"
  },
  navItemTextActive: {
    color: "#182C4F"
  },
  badge: {
    right: -60,
    padding: "0px 7px",
    backgroundColor: "#0C66FF"
  },
  logoutIcon: {
    width: 26,
    height: 26,
    background: "#999",
    borderRadius: "50%",
    display: "flex",
    marginRight: 10,
    justifyContent: "center",
    alignItems: "center"
  }
});
