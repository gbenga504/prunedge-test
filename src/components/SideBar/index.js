import React from "react";
import withStyles from "@material-ui/core/styles/withStyles";
import Typography from "@material-ui/core/Typography";
import HomeIcon from "@material-ui/icons/Home";
import Badge from "@material-ui/core/Badge";
import ArrowIcon from "@material-ui/icons/KeyboardBackspace";
import BookmarkIcon from "@material-ui/icons/Bookmark";
import ActivityIcon from "@material-ui/icons/NotificationImportant";
import ArchiveIcon from "@material-ui/icons/Archive";
import SettingsIcon from "@material-ui/icons/SettingsApplications";
import classNames from "classnames";

import { styles } from "./styles";

class SideBar extends React.PureComponent {
  state = {
    activeNavItem: "settings"
  };

  handleSetActiveNavItem = activeNavItem => {
    this.setState({
      activeNavItem
    });
  };

  navItems = [
    { name: "dashboard", icon: HomeIcon },
    { name: "documents", icon: BookmarkIcon },
    { name: "activity", icon: ActivityIcon },
    { name: "archive", icon: ArchiveIcon },
    { name: "settings", icon: SettingsIcon }
  ];

  renderUserDetails = () => {
    let { classes } = this.props;

    return (
      <div className={classes.userDetailsContainer}>
        <div className={classes.avatarContainer}>
          <img
            alt="user"
            src="https://media.licdn.com/dms/image/C5603AQHnNE6t21CVkg/profile-displayphoto-shrink_100_100/0?e=1571875200&v=beta&t=DOFMyno8Ei2tZmecWIppdRxDcicR43lX-0k2Q7rktaU"
          />
        </div>
        <Typography variant="body2" classes={{ body2: classes.userName }}>
          Jaohn Samuel
        </Typography>
        <Typography variant="body2" classes={{ body2: classes.role }}>
          Front Desk Officer
        </Typography>
        <Typography variant="body2" classes={{ body2: classes.department }}>
          Finance
        </Typography>
      </div>
    );
  };

  renderNavItems = () => {
    let { classes } = this.props,
      { activeNavItem } = this.state;

    return (
      <div style={{ display: "flex", flexDirection: "column", marginTop: 38 }}>
        {this.navItems.map(navItem => {
          let Icon = navItem.icon;

          return (
            <div
              key={navItem.name}
              className={classNames(
                classes.navItem,
                activeNavItem === navItem.name && classes.navItemActive
              )}
              onClick={() => this.handleSetActiveNavItem(navItem.name)}
            >
              <Icon
                className={classNames(
                  classes.icon,
                  activeNavItem === navItem.name && classes.iconActive
                )}
              />
              <Typography
                variant="body2"
                classes={{
                  body2: classNames(
                    classes.navItemText,
                    activeNavItem === navItem.name && classes.navItemTextActive
                  )
                }}
              >
                {navItem.name}
              </Typography>
              {navItem.name === "activity" && (
                <Badge
                  badgeContent={12}
                  color="primary"
                  classes={{ badge: classes.badge }}
                />
              )}
            </div>
          );
        })}
      </div>
    );
  };

  renderTopSection = () => {
    let { classes } = this.props;
    return (
      <div className={classes.topSectionContainer}>
        {this.renderUserDetails()}
        {this.renderNavItems()}
      </div>
    );
  };

  render() {
    let { classes } = this.props;

    return (
      <div className={classes.container}>
        {this.renderTopSection()}
        <div className={classes.navItem}>
          <div className={classes.logoutIcon}>
            <ArrowIcon style={{ fontSize: 15, color: "#fff" }} />
          </div>
          <Typography
            variant="body2"
            classes={{ body2: classes.navItemText }}
          >
            Logout
          </Typography>
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(SideBar);
