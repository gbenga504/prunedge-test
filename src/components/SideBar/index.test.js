import React from "react";
import { create } from "react-test-renderer";

import SideBar from "./index";

describe("SideBar", () => {
  it("renders the sidebar", () => {
    const tree = create(<SideBar />);
    expect(tree).toMatchSnapshot();
  });
});
