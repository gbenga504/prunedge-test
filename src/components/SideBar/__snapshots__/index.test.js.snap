// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders the sidebar 1`] = `
<div
  className="SideBar-container-1"
>
  <div
    className="SideBar-topSectionContainer-2"
  >
    <div
      className="SideBar-userDetailsContainer-3"
    >
      <div
        className="SideBar-avatarContainer-4"
      >
        <img
          alt="user"
          src="https://media.licdn.com/dms/image/C5603AQHnNE6t21CVkg/profile-displayphoto-shrink_100_100/0?e=1571875200&v=beta&t=DOFMyno8Ei2tZmecWIppdRxDcicR43lX-0k2Q7rktaU"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 SideBar-userName-5"
      >
        Jaohn Samuel
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 SideBar-role-6"
      >
        Front Desk Officer
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 SideBar-department-7"
      >
        Finance
      </p>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "marginTop": 38,
        }
      }
    >
      <div
        className="SideBar-navItem-8"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root SideBar-icon-10"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 SideBar-navItemText-12"
        >
          dashboard
        </p>
      </div>
      <div
        className="SideBar-navItem-8"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root SideBar-icon-10"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 SideBar-navItemText-12"
        >
          documents
        </p>
      </div>
      <div
        className="SideBar-navItem-8"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root SideBar-icon-10"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 16v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-5 0h-2v-2h2v2zm0-4h-2V8h2v4zm-1 10c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 SideBar-navItemText-12"
        >
          activity
        </p>
        <span
          className="MuiBadge-root"
        >
          <span
            className="MuiBadge-badge SideBar-badge-14 MuiBadge-colorPrimary"
          >
            12
          </span>
        </span>
      </div>
      <div
        className="SideBar-navItem-8"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root SideBar-icon-10"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M20.54 5.23l-1.39-1.68C18.88 3.21 18.47 3 18 3H6c-.47 0-.88.21-1.16.55L3.46 5.23C3.17 5.57 3 6.02 3 6.5V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V6.5c0-.48-.17-.93-.46-1.27zM12 17.5L6.5 12H10v-2h4v2h3.5L12 17.5zM5.12 5l.81-1h12l.94 1H5.12z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 SideBar-navItemText-12"
        >
          archive
        </p>
      </div>
      <div
        className="SideBar-navItem-8 SideBar-navItemActive-9"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root SideBar-icon-10 SideBar-iconActive-11"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm7-7H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-1.75 9c0 .23-.02.46-.05.68l1.48 1.16c.13.11.17.3.08.45l-1.4 2.42c-.09.15-.27.21-.43.15l-1.74-.7c-.36.28-.76.51-1.18.69l-.26 1.85c-.03.17-.18.3-.35.3h-2.8c-.17 0-.32-.13-.35-.29l-.26-1.85c-.43-.18-.82-.41-1.18-.69l-1.74.7c-.16.06-.34 0-.43-.15l-1.4-2.42c-.09-.15-.05-.34.08-.45l1.48-1.16c-.03-.23-.05-.46-.05-.69 0-.23.02-.46.05-.68l-1.48-1.16c-.13-.11-.17-.3-.08-.45l1.4-2.42c.09-.15.27-.21.43-.15l1.74.7c.36-.28.76-.51 1.18-.69l.26-1.85c.03-.17.18-.3.35-.3h2.8c.17 0 .32.13.35.29l.26 1.85c.43.18.82.41 1.18.69l1.74-.7c.16-.06.34 0 .43.15l1.4 2.42c.09.15.05.34-.08.45l-1.48 1.16c.03.23.05.46.05.69z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 SideBar-navItemText-12 SideBar-navItemTextActive-13"
        >
          settings
        </p>
      </div>
    </div>
  </div>
  <div
    className="SideBar-navItem-8"
  >
    <div
      className="SideBar-logoutIcon-15"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        style={
          Object {
            "color": "#fff",
            "fontSize": 15,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
        />
      </svg>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 SideBar-navItemText-12"
    >
      Logout
    </p>
  </div>
</div>
`;
